test/puppetlabs/ssl_utils/core_test.clj

```diff
@@ -500,6 +501,60 @@
+         cert-with-expired-crl (-> "crl_validation/cert-with-expired-crl.pem" open-ssl-file pem->certs)
+         expired-crl (-> "crl_validation/expired-crl.pem" open-ssl-file pem->crls)
+         cert-with-not-valid-crl (-> "crl_validation/cert-with-not-valid-crl.pem" open-ssl-file pem->certs)
+         not-yet-valid-crl (-> "crl_validation/not-yet-valid-crl.pem" open-ssl-file pem->crls)
+         cert-chain-with-crl-bad-sig (-> "crl_validation/cert-with-crl-bad-sig.pem" open-ssl-file pem->certs)
+         crl-chain-bad-signature (-> "crl_validation/crl-with-bad-signature.pem" open-ssl-file pem->crls)
+         cert-chain-with-revoked-cert (-> "crl_validation/cert-chain-with-revoked-cert.pem" open-ssl-file pem->certs)
+         crl-chain-cert-revoked (-> "crl_validation/crl-chain-with-cert-revoked.pem" open-ssl-file pem->crls)
+         ;;crl-chain-delta (open-ssl-file "crl_validation/3-crl-chain-delta.pem")
+         ]
+     (testing "valid certs and CRL chain"
+       (is (nil? (validate-cert-chain cert-chain crl-chain))))
+     (testing "extra CRLs are okay"
+       (is (nil? (validate-cert-chain cert-chain crl-chain-extra-crls))))
+     (testing "missing CRLs are rejected"
+       ;; currently UNDETERMINED REVOCATION STATUS
+       (is (thrown-with-msg? CertPathValidatorException
+                             #"Could not determine revocation status"
+                             (validate-cert-chain cert-chain crl-chain-missing-crls))))
+     (testing "totally unrelated CRLs are rejected"
+       ;; currently UNDETERMINED REVOCATION STATUS
+       (is (thrown-with-msg? CertPathValidatorException
+                             #"Could not determine revocation status"
+                             (validate-cert-chain cert-chain other-crl-chain))))
+     (testing "expired CRLs are rejected"
+       ;; currently UNDETERMINED REVOCATION STATUS
+       (is (thrown-with-msg? CertPathValidatorException
+                             #"Could not determine revocation status"
+                             (validate-cert-chain cert-with-expired-crl expired-crl))))
+     (testing "not-yet-valid CRLs are rejected"
+       ;; currently UNDETERMINED REVOCATION STATUS
+       (is (thrown-with-msg? CertPathValidatorException
+                             #"Could not determine revocation status"
+                             (validate-cert-chain cert-with-not-valid-crl not-yet-valid-crl))))
+     (testing "CRLs with bad signatures (aka not signed by their issuer) are rejected"
+       ;; I might want to call this something other than "bad signature",
+       ;; not sure if that's descriptive
+       ;; currently UNDETERMINED REVOCATION STATUS
+       (is (thrown-with-msg? CertPathValidatorException
+                             #"Could not determine revocation status"
+                             (validate-cert-chain cert-chain-with-crl-bad-sig crl-chain-bad-signature))))
+     (testing "rejected when cert in chain has been revoked"
```

Comment:
We should be able to catch this scenario when we call it in the CA service, since it has a different Reason than the other failure cases, but I think it's still appropriate to throw the error in this function.

Reply:
Seems like there was a little bit of disagreement in the ticket between stakeholders as to whether we should honor this case, too.